test/CMakeLists.txt

```diff
@@ -325,6 +326,16 @@ itk_add_test(NAME itkInverse1DFFTImageFilterTest
     ${CMAKE_CURRENT_SOURCE_DIR}/Input/itkForward1DFFTImageFilterTestBaseline
     ${ITK_TEST_OUTPUT_DIR}/itkInverse1DFFTImageFilterTestOutput.mha
     )
+
+itk_add_test(NAME itkSpeckleReducingAnisotropicDiffusionImageFilterTest
+  COMMAND UltrasoundTestDriver
+  --compare
+  DATA{Baseline/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutputBaseline.mha}
+  ${ITK_TEST_OUTPUT_DIR}/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutput.mha
+  itkSpeckleReducingAnisotropicDiffusionImageFilterTest
+  DATA{Input/PhantomRFFrame0.mha}
+  ${ITK_TEST_OUTPUT_DIR}/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutput.mha
+  )
```

Comment:
Is there a comparison issue here that I'm not seeing? CTest output appears to match expected paths:

> 34: Test command: C:\repos\ITK-build-unstable\Wrapping\Generators\Python\itk\UltrasoundTestDriver.exe "--compare" "C:/repos/unstable-us/ExternalData/test/Baseline/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutputBaseline.mha" "C:/repos/unstable-us/Testing/Temporary/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutput.mha" "itkSpeckleReducingAnisotropicDiffusionImageFilterTest" "C:/repos/unstable-us/ExternalData/test/Input/PhantomRFFrame0.mha" "C:/repos/unstable-us/Testing/Temporary/itkSpeckleReducingAnisotropicDiffusionImageFilterTestOutput.mha"

Reply:
You might need to clean the various versions of this file in your build tree.

Reply:
Running a clean build did not resolve the issue. Steps to reproduce:

1. Do a clean build of ITKUltrasound (specifically UltrasoundTestDriver)
2. Change a parameter in the denoising filter. I reduced the number of iterations to 1 so that very little denoising occurs.
3. Run itkSpeckleReducingAnisotropicDiffusion CTest (passes)
4. Visually compare test output to baseline

Expected: Output image is different from baseline and CTest comparison fails
Observed: Output image is different from baseline but CTest comparison succeeds

I have verified that the output image is updated at the time of the test and that the baseline image path logged by CTest matches the expected baseline image.

@dzenanz @thewtex Do you have other thoughts as to what might be going wrong here? If preferred I can document with an Issue and go ahead and complete this PR.

Reply:
Intensity tolerance is 2.0 by default:
https://github.com/InsightSoftwareConsortium/ITK/blob/8be208ee1f04d09912523401e78ba7bb14a89d6f/Modules/Core/TestKernel/src/itkTestDriverInclude.cxx#L124-L125
If your pixels have 0.0 to 1.0 range, that would explain it.

Reply:
This was the answer, thank you @dzenanz ! Pixel values range from ~0.0 to 2.0 in the input and baseline images. Reducing to 0.01 and running a single denoising iteration resulted in an error as expected when comparing with the baseline, while running 100 denoising to generate the baseline image again resulted in a passing test.

I wasn't aware of the variety of test arguments before this, I will keep them in mind for future issues.